yt/yt/server/master/object_server/object_service.cpp

```diff
@@ -1659,16 +1661,13 @@ class TObjectService::TExecuteSession
 
             auto throttlerFuture = securityManager->ThrottleUser(User_.Get(), 1, WorkloadType);
 
-            if constexpr (SubrequestType == EExecutionSessionSubrequestType::LocalWrite) {
+            if constexpr (SubrequestType == EExecutionSessionSubrequestType::LocalRead) {
                 if (User_.Get() != securityManager->GetRootUser()) {
-                    // We chain futures like this to avoid dealing with of ownership of the underlying promise of an AllSet combiner.
-                    // Note that the strong ref to the session ends up somewhere within the user throttler object.
-                    // Also keep in mind that we must obtain the user throttler future here, and not in a an arbitrary callback thread,
-                    // because we can only dereference the user pointer from a persistent-state-read thread.
-                    throttlerFuture = throttlerFuture.Apply(
-                        BIND_NO_PROPAGATE([localWriteThrottlerFuture = Owner_->LocalWriteRequestThrottler_->Throttle(1)] {
-                            return localWriteThrottlerFuture;
-                        }));
+                    throttlerFuture = AllSucceeded(std::vector{throttlerFuture, Owner_->LocalReadRequestThrottler_->Throttle(1)});
+                }
+            } else if constexpr (SubrequestType == EExecutionSessionSubrequestType::LocalWrite) {
+                if (User_.Get() != securityManager->GetRootUser()) {
+                    throttlerFuture = AllSucceeded(std::vector{throttlerFuture, Owner_->LocalWriteRequestThrottler_->Throttle(1)});
```

Comment:
AllSucceeded takes `TFutureCombinerOptions`. Let's disable both [cancellation mechanics](https://github.com/ytsaurus/ytsaurus/blob/a4a378f7057a992d5db119b0438682672702a333/yt/yt/core/actions/future.h#L641-L651) for this case. We are not prepared for any cancellations here anywhere, so let's guard against it.